platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -255,55 +244,51 @@ message RoundRosterPair {
+Roster invalidates and removes the current candidate roster. Therefore, the map is expected to have a maximum size
+of three elements.
+
 The new `RosterState` will store the `candidateRosterHash` and a list `roundRosterPairs`
 of pairs of round numbers and the active roster that was used for that round. The `roundRosterPairs` list will be
-updated only when the active roster changes.
-In the vast majority of cases, the list will contain only one element, the current active roster.
-Immediately following the adoption of a candidate roster, this list will have two elements — the previous active roster
-and the current active roster (which was the candidate roster).
-The choice of a list is suitable here as it is much cheaper to iterate over 2-3 elements comparing their `long` round
-numbers than it is to do a hash lookup.
-When we reach full Dynamic Address Book, this design choice may need to be revisited.
+updated only when the active roster changes. In the vast majority of cases, the list will contain only one element,
+the current active roster.
 
-If a `candidateRosterHash` hash entry already exist (hash collisions) in the map of Rosters, it will be discarded. That
-is, setting a candidate roster is an idempotent operation.
+Immediately following the adoption of a candidate roster, this list will have two elements — the previous active roster
+and the current active roster (which was the candidate roster). The choice of a list is suitable here as it is much
+cheaper to iterate over 2-3 elements comparing their `long` round numbers than it is to do a hash lookup. When we
+reach full Dynamic Address Book, this design choice may need to be revisited.
 
-One benefit of this indirection (using a Map instead of a Singleton) is that it avoids moving data around in the merkle
-state.
-Another benefit is that adoption trigger becomes straightforward (App sets the roster) with delineated
-responsibilities between the App and the platform.
+If a `candidateRosterHash` hash entry already exist (hash collisions) in the map of Rosters, it will be discarded.
+That is, setting a candidate roster is an idempotent operation. One benefit of this indirection (using a Map
+instead of a Singleton) is that it avoids moving data around in the merkle state. Another benefit is that adoption
+trigger becomes straightforward (App sets the roster) with delineated responsibilities between the App and the platform.
 
 ### Roster Validity
 
 A Roster is considered valid if it satisfies the following conditions:
 
 1. The roster must have at least one RosterEntry.
 2. At least one RosterEntry/ies must have a non-zero weight.
-3. All RosterEntry/ies must have a valid X509Certificate.
-4. All RosterEntry/ies must have a valid PairingPublicKey.
+3. All RosterEntry/ies must have a valid gossip_ca_certificate which is a DER encoded X509Certificate.
+4. All RosterEntry/ies must have a valid tss_encryption_key which is a DER encoded X509Certificate.
```

Comment:
The tss_encryption_key isn't a DER encoded X509Certificate. To avoid duplicating the format descriptions, I suggest to rephrase the validation checks 3 and 4 to simply refer to the protobuf specifications where the format is actually specified.